File: src/main/java/org/jukeboxmc/form/Modal.java
package org.jukeboxmc.form;

import org.jetbrains.annotations.NotNull;
import org.json.simple.JSONObject;

/**
 * @author GoMint
 * @version 1.0
 */
public class Modal extends Form<Boolean> {

    private final String question;
    private String trueButtonText;
    private String falseButtonText;

    public Modal( String title, String question ) {
        super( title );
        this.question = question;
    }

    public void setTrueButtonText( String text ) {
        this.trueButtonText = text;
        this.dirty = true;
    }

    public void setFalseButtonText( String text ) {
        this.falseButtonText = text;
        this.dirty = true;
    }

    public @NotNull String getFormType() {
        return "modal";
    }

    @Override
    public @NotNull JSONObject toJSON() {
        // Fast out when cached
        if ( this.cache != null && !this.dirty ) {
            return this.cache;
        }

        // Create new JSON view of this form
        JSONObject jsonObject = super.toJSON();
        jsonObject.put( "content", this.question );
        jsonObject.put( "button1", this.trueButtonText );
        jsonObject.put( "button2", this.falseButtonText );

        // Cache and return
        this.cache = jsonObject;
        this.dirty = false;
        return this.cache;
    }

    @Override
    public @NotNull Boolean parseResponse(@NotNull String json ) {
        return json.trim().equals( "true" );
    }

}

File: src/main/java/org/jukeboxmc/form/Modal.kt
package org.jukeboxmc.form

import com.fasterxml.jackson.annotation.JsonProperty

/**
 * @author GoMint
 * @version 1.0
 */
class Modal(
    title: String,
    @get:JsonProperty("content")
    val question: String,
) : Form<Boolean>(title) {
    @JsonProperty("button1")
    var trueButtonText: String? = null

    @JsonProperty("button2")
    var falseButtonText: String? = null

    override val formType: String
        get() = "modal"

    override fun parseResponse(json: String): Boolean {
        return json.trim { it <= ' ' } == "true"
    }
}
